docs/source/en/api/pipelines/unidiffuser.mdx

```diff
@@ -0,0 +1,215 @@
+image_url = "https://huggingface.co/datasets/hf-internal-testing/diffusers-images/resolve/main/unidiffuser/unidiffuser_example_image.jpg"
+response = requests.get(image_url)
+init_image = Image.open(BytesIO(response.content)).convert("RGB")
+init_image = init_image.resize((512, 512))
+
+sample = pipe(image=init_image, num_inference_steps=20, guidance_scale=8.0)
+i2t_text = sample.text[0]
+print(text)
+```
+
+The `img2text` mode requires that an input `image` be supplied. You can set the `img2text` mode manually with [`UniDiffuser.set_image_to_text_mode`].
+
+### Image Variation
+
+The UniDiffuser authors suggest performing image variation through a "round-trip" generation method, where given an input image, we first perform an image-to-text generation, and the perform a text-to-image generation on the outputs of the first generation.
+This produces a new image which is semantically similar to the input image:
+
+```python
+import requests
+import torch
+from PIL import Image
+from io import BytesIO
+
+from diffusers import UniDiffuserPipeline
+
+device = "cuda"
+model_id_or_path = "thu-ml/unidiffuser-v1"
+pipe = UniDiffuserPipeline.from_pretrained(model_id_or_path, torch_dtype=torch.float16)
+pipe.to(device)
+
+# Image variation can be performed with a image-to-text generation followed by a text-to-image generation:
+# 1. Image-to-text generation
+image_url = "https://huggingface.co/datasets/hf-internal-testing/diffusers-images/resolve/main/unidiffuser/unidiffuser_example_image.jpg"
+response = requests.get(image_url)
+init_image = Image.open(BytesIO(response.content)).convert("RGB")
+init_image = init_image.resize((512, 512))
+
+sample = pipe(image=init_image, num_inference_steps=20, guidance_scale=8.0)
+i2t_text = sample.text[0]
+print(text)
+
+# 2. Text-to-image generation
+sample = pipe(prompt=i2t_text, num_inference_steps=20, guidance_scale=8.0)
+final_image = sample.images[0]
+final_image.save("unidiffuser_image_variation_sample.png")
+```
+
+### Text Variation
+
+
+Similarly, text variation can be performed on an input prompt with a text-to-image generation followed by a image-to-text generation:
+
+```python
+import torch
+
+from diffusers import UniDiffuserPipeline
+
+device = "cuda"
+model_id_or_path = "thu-ml/unidiffuser-v1"
+pipe = UniDiffuserPipeline.from_pretrained(model_id_or_path, torch_dtype=torch.float16)
+pipe.to(device)
+
+# Text variation can be performed with a text-to-image generation followed by a image-to-text generation:
+# 1. Text-to-image generation
+prompt = "an elephant under the sea"
+
+sample = pipe(prompt=prompt, num_inference_steps=20, guidance_scale=8.0)
+t2i_image = sample.images[0]
+t2i_image.save("unidiffuser_text2img_sample_image.png")
+
+# 2. Image-to-text generation
+sample = pipe(image=t2i_image, num_inference_steps=20, guidance_scale=8.0)
+final_prompt = sample.text[0]
+print(final_prompt)
+```
+
+## UniDiffuserPipeline
+[[autodoc]] UniDiffuserPipeline
+	- all
+	- __call__
+
+## ImageTextPipelineOutput
```

Comment:
This should go here: 

https://github.com/huggingface/diffusers/blob/main/docs/source/en/api/outputs.mdx

Reply:
Also wanted to know if there's any argument control the number of images / text I wanted to generate as a part of the variation mode.

Reply:
I can control the `num_images_per_prompt` in the text-to-image mode, so that's settled. But what about text variation?

Reply:
For modes which generate only text (`img2text` and `text`), there's an analogous `num_prompts_per_image` argument to `__call__ `. So when you perform the second `img2text` generation for text variation you can specify `num_prompts_per_image > 1` to get multiple text variation samples.

Reply:
Thanks!

Reply:
> This should go here:
> 
> https://github.com/huggingface/diffusers/blob/main/docs/source/en/api/outputs.mdx

It feels more natural to me to have the documentation for `ImageTextPipelineOutput` alongside `ImagePipelineOutput`, which is at the [`Diffusion Pipeline`](https://huggingface.co/docs/diffusers/main/en/api/diffusion_pipeline#diffusers.ImagePipelineOutput) doc page.

Reply:
I've gone ahead and moved the `ImageTextPipelineOutput` documentation to `/api/diffusion_pipeline.mdx` (alongside the `ImagePipelineOutput` and `AudioPipelineOutput` documentation). Let me know if it would be better somewhere else (for example, at `/api/outputs.mdx` as originally suggested) :).

Reply:
I understand. But we will soon update that too :)

Reply:
I see, so should I move it to `/api/outputs`? Or is it fine to leave it at `/api/diffusion_pipeline` for now?